src/nccl_ofi_idpool.cpp

```diff
@@ -4,116 +4,70 @@
 
 #include "config.h"
 
-#include <assert.h>
-#include <errno.h>
-#include <stdlib.h>
+#include <mutex>
+#include <vector>
+
+#include <unistd.h>
```

Comment:
In the testing for this PR, I had ran into an issue where `ssize_t` was not defined on Ubuntu 22.04 without including `unistd.h` despite it being defined on AL2/AL2023. Given this, would it be appropriate for me to still include `errno.h` for the error codes and  `stdint.h` for `uint64_t` in case an OS would not have those definitions by default?

Reply:
Looking at the `ssize_t` build failure again [here](https://github.com/aws/aws-ofi-nccl/actions/runs/14180354608/job/39724851231), the difference for whether `ssize_t` was defined without including `unistd.h` was not between Ub22 vs AL2/AL2023, and instead was between which compiler and compiler version was used on Ub22 (`gcc-latest` failed the build, while `gcc-legacy`/`clang-legacy`/`clang-latest` succeeded).